optuna_dashboard/ts/components/GraphEdf.tsx

```diff
@@ -28,44 +12,16 @@ interface EdfPlotInfo {
 }
 
 export const GraphEdf: FC<{
-  study: StudyDetail | null
+  studies: StudyDetail[]
   objectiveId: number
-}> = ({ study, objectiveId }) => {
+}> = ({ studies, objectiveId }) => {
   const theme = useTheme()
   const domId = getPlotDomId(objectiveId)
   const target = useMemo<Target>(
     () => new Target("objective", objectiveId),
     [objectiveId]
   )
-  const trials = useFilteredTrials(study, [target], false)
-
-  useEffect(() => {
-    if (study !== null) {
-      plotEdf(trials, target, domId, theme.palette.mode)
-    }
-  }, [trials, target, domId, theme.palette.mode])
-  return (
-    <Box>
-      <Typography
-        variant="h6"
-        sx={{ margin: "1em 0", fontWeight: theme.typography.fontWeightBold }}
-      >
-        {`EDF for ${target.toLabel(study?.objective_names)}`}
-      </Typography>
-      <Box id={domId} sx={{ height: "450px" }} />
-    </Box>
-  )
-}
-
-export const GraphEdfMultiStudies: FC<{
-  studies: StudyDetail[]
-}> = ({ studies }) => {
-  const theme = useTheme()
-  const [targets, selected, setTarget] = useObjectiveTargets(
-    studies.length !== 0 ? studies[0] : null
-  )
-
-  const trials = useFilteredTrialsFromStudies(studies, [selected], false)
+  const trials = useFilteredTrialsFromStudies(studies, [target], false)
```

Comment:
Actually I have some concerns about the current implementation of `useFilteredTrialsFromStudies(studies, [target], false)` as commented in https://github.com/optuna/optuna-dashboard/pull/380#discussion_r1138403763.

Does it sound reasonable to fix the logic of `useFilteredTrialsFromStudies` first? Then we can address the refactoring of `GraphEdf`.

Reply:
If I understand it correctly, [this comment](https://github.com/optuna/optuna-dashboard/pull/380#discussion_r1138403763) is only applied to GraphHistory because GraphEdf does not use `useObjectiveAndUserAttrTargets`. Does I misunderstand anything?